cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -575,6 +575,15 @@ Future<Void> rollingUpdateForNewCaKey() {
                 false,
                 eventPublisher
         ).rollingRestart(pod -> {
+            if (podRollReasons.getReasons().size() == 1 && podRollReasons.contains(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED)) {
```

Comment:
Why is this needed? Why do we even need the rolling update here for Clients CA key replacement?
* It is not used internally, so we do not need to establish trust
* It should be enough to roll this in the regular rolling update where it would anyway roll to change the annotation, or?